examples/morpheusvm/tests/integration/integration_test.go

```diff
@@ -212,7 +215,40 @@ var _ = ginkgo.BeforeSuite(func() {
 		toEngine := make(chan common.Message, 1)
 		db := memdb.New()
 
-		v, err := controller.New(vm.WithManualGossiper(), vm.WithManualBuilder())
+		server, handler := rpc.NewWebSocketServer(
+			l,
+			trace.Noop,
+			registry.Action,
+			registry.Auth,
+			10000000,
+		)
+
+		webSocketFactory := rpc.NewWebSocketServerFactory(handler)
+		removeTxSubscription := rpc.SubscriptionFuncFactory[vm.TxRemovedEvent]{
+			AcceptF: func(event vm.TxRemovedEvent) error {
+				return server.RemoveTx(event.TxID, event.Err)
+			},
+		}
+
+		acceptBlockSubscription := &rpc.SubscriptionFuncFactory[*chain.StatelessBlock]{
+			AcceptF: func(event *chain.StatelessBlock) error {
+				return server.AcceptBlock(event)
+			},
+		}
+
+		v, err := controller.New(
+			vm.WithManualGossiper[*controller.Controller](),
+			vm.WithManualBuilder[*controller.Controller](),
```

Comment:
We need `WithManualGossiper` and `WithManualBuilder` here as overrides for the integration tests, but can we move the others into `controller.New(...)` itself, so that we don't need to place them here and in the entrypoint to the VM?

Reply:
This creates an annoying circular dependency between the morpheusvm `controller` and `rpc` packages (since now the controller consumes the rpc package to initialize the default api, but the rpcs also need to know the controller to expose the api on the concrete type). I'm going to fix this by just merging the two packages together since the controller at this point is logically just controller api + genesis/upgrade and we will have a follow-up to merge the controller-defined apis into the VM apis since the controller apis right now are in a weird spot where they serve hypersdk-defined data like balances.

Reply:
Sounds good